packages/components/data-table/src/header-cell.styles.tsx

```diff
@@ -100,12 +152,15 @@ const BaseHeaderCell = styled.th<TBaseHeaderCell>`
 
   /* remove user-agent styles */
   padding: 0;
-  font-weight: ${designTokens.fontWeightForTableHeader};
-  font-size: ${designTokens.fontSizeForTable};
+  font-weight: ${designTokens.fontWeight500};
+  font-size: ${designTokens.fontSize10};
 
   /* right border that doesn't count towards the column width */
   box-shadow: inset -1px 0 ${designTokens.borderColorForTableHeader};
 
+  border-bottom: ${designTokens.borderWidth1} solid
+    ${designTokens.colorNeutral95};
```

Comment:
I've observed this effect:
<img width="205" alt="Screenshot 2023-10-20 at 10 48 50" src="https://github.com/commercetools/ui-kit/assets/49066275/c22c6097-73ed-446d-956d-edece053eed9">
Just thought I'd suggest it as a possible alternative:
```suggestion
  /**
   * bottom border that doesn't overshadow the resize indicator
   */
  :after {
    content: '';
    position: absolute;
    z-index: -1;
    width: 100%;
    height: ${designTokens.borderWidth1};
    bottom: 0;
    left: 0;
    background-color: ${designTokens.colorNeutral95};
  }
```